velox/exec/tests/RowContainerTest.cpp

```diff
@@ -1266,4 +1266,46 @@ TEST_F(RowContainerTest, unknown) {
     ASSERT_TRUE(rowContainer->equals<false>(
         rows[row], rowContainer->columnAt(0), decoded, row));
   }
+
+  {
+    // Verify compare method with two rows as input
+    std::sort(rows.begin(), rows.end(), [&](const char* l, const char* r) {
+      return rowContainer->compare(l, r, 0, {}) < 0;
+    });
+    VectorPtr result = BaseVector::create(UNKNOWN(), rows.size(), pool_.get());
+    rowContainer->extractColumn(rows.data(), rows.size(), 0, result);
+    // Since the Vector is just a null constant, it should be indistinguishable
+    // from the original.
+    assertEqualVectors(data->childAt(0), result);
+  }
+
+  std::vector<std::pair<int, char*>> indexedRows(rows.size());
+  for (size_t i = 0; i < rows.size(); ++i) {
+    indexedRows[i] = std::make_pair(i, rows[i]);
+  }
+
+  // Verify compare method with row and decoded vector as input
+  {
+    std::sort(
+        indexedRows.begin(),
+        indexedRows.end(),
+        [&](const std::pair<int, char*>& l, const std::pair<int, char*>& r) {
+          return rowContainer->compare(
+                     l.second,
+                     rowContainer->columnAt(0),
+                     decoded,
+                     r.first,
+                     {}) < 0;
+        });
+  }
```

Comment:
do we need to create a vector from the sorted indexRows and compare with an expected vector? Here and for the case below?

Reply:
I didn't because the vector is just a NULL constant vector, regardless of the order of indexRows it's still going to be a NULL constant vector.

Reply:
I see, thanks for clarifying, I guess from the code it was not apparent what we were testing here by calling sort on the data. what happens if i run this test without your change? if it would throw an exception or cause a segfault, but with this change it just works, maybe we can add a check for that by calling VELOX_NO_THROW or similar to make sure we catch it if things change in the future.